src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -296,7 +296,10 @@ private void EnsureServer()
             }
 
             // Fire the IHostedService.Stop
-            _hostedServiceExecutor?.Stop();
+            if (_hostedServiceExecutor != null)
```

Comment:
For whatever reason I saw _hostedServiceExecutor being set in the ctor not in StartAsync. Need new glasses.